appsec/src/extension/request_lifecycle.c

```diff
@@ -128,8 +128,19 @@ static zend_array *nullable _do_request_begin_user_req(zval *nullable rbe_zv)
     return _do_request_begin(rbe_zv, true);
 }
 
-static bool _rem_cfg_path_changed()
+static bool _has_empty_service_or_env()
 {
+    return zend_string_equals_literal(get_DD_ENV(), "") ||
+           zend_string_equals_literal(get_DD_SERVICE(), "");
+}
```

Comment:
You should cache that at rinit, it's possible to change these via `ini_set()` too at runtime... Which will then not reflect the truth for you in rshtudown.

Reply:
Can't I use `get_global_DD_ENV()` here for the same effect?

Reply:
I've moved it to startup with get_global

Reply:
No, get_global is definitely inferior. It will also bypass all perdir changes. Like vhost specific config.

Reply:
Just cache the value in a thread-local variable after rinit.